GenericApp/app/src/main/java/io/openremote/app/ui/HostSelectionFragment.kt

```diff
@@ -50,8 +51,12 @@ class HostSelectionFragment : Fragment() {
 
     private fun connectToHost(host: String) {
         parentActivity.binding.progressBar.visibility = View.VISIBLE
-        val url =
-            if (URLUtil.isValidUrl(host)) host.plus("/api/master") else "https://${host}.openremote.app/api/master"
+        val url = when {
+            URLUtil.isValidUrl(host) -> host.plus("/api/master")
+            UrlUtils.isIpAddress(host) -> "https://${host}/api/master"
+            !UrlUtils.startsWithHttp(host) && UrlUtils.endsWithTld(host) -> "https://${host}/api/master"
```

Comment:
host already contains http or https, shouldn't we return to ${host}/api/master ?

Reply:
The first check already checks if it contains http or https, so if that's false, it will continue, meaning nor http nor https is present

Reply:
@ebariaux are you ok with this now?

Reply:
My bad, I missed the ! at the beginning of the line, so yes, that's OK

Reply:
So you can approve? ;-)